src/impl_/coroutine.rs

```diff
@@ -56,7 +56,11 @@ impl<T: PyClass> Deref for RefGuard<T> {
 
 impl<T: PyClass> Drop for RefGuard<T> {
     fn drop(&mut self) {
-        Python::with_gil(|gil| self.0.as_ref(gil).release_ref())
+        Python::with_gil(|gil| {
+            #[allow(deprecated)]
+            let self_ref = self.0.as_ref(gil);
+            self_ref.release_ref()
```

Comment:
Potentially the solution here is to move `.release_ref()` and `.release_mut()` to be crate-private methods on `Bound<T: PyClass>`.

Reply:
Do you have any suggestions what that should look like? I don't understand what the current implementation is doing.

Reply:
Maybe a simpler option just for the moment would be to make the methods on `Bound` just call the methods on `PyCell` (rather than move them), and then when `Bound::get_cell` is cleaned up we can clean those at the same time?